flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/utils/TypeTransformer.java

```diff
@@ -675,4 +686,37 @@ private String getTypeName(Symbol symbol) {
     private List<String> getTypeRefs(Object type, TypeSymbol typeDescriptor) {
         return type instanceof String ? TypeUtils.getTypeRefIds(typeDescriptor, moduleInfo) : List.of();
     }
+
+    private Object handleAsFirstClassNonIntersectionType(IntersectionTypeSymbol intersectionTypeSymbol,
+                                                         TypeData.TypeDataBuilder typeDataBuilder) {
+        SemanticModel semanticModel = module.getCompilation().getSemanticModel();
+        Types types = semanticModel.types();
+
+        TypeSymbol nonReadonlyTypeSymbol = null;
+        List<TypeSymbol> intersectionMemberTypes = intersectionTypeSymbol.memberTypeDescriptors();
+
+        // Check for non-readonly type members to treat the type as a first-class non-intersection type
+        for (TypeSymbol typeSymbol : intersectionMemberTypes) {
+            if (typeSymbol.subtypeOf(types.READONLY)) {
+                continue;
+            }
+
+            if (nonReadonlyTypeSymbol == null) {
+                nonReadonlyTypeSymbol = typeSymbol;
+            } else {
+                // If there are multiple non-readonly types, we cannot handle it as a first-class non-intersection type
+                return transform(intersectionTypeSymbol, typeDataBuilder);
+            }
+        }
+
+        if (nonReadonlyTypeSymbol == null || nonReadonlyTypeSymbol.typeKind() == TypeDescKind.TYPE_REFERENCE) {
+            // If no non-readonly type is found or the found non-readonly type is a type-reference type,
```

Comment:
```suggestion
            // If no non-readonly type is not found or the found non-readonly type is a type-reference type,
```

Reply:
"If **no** non-readonly type is found"
.    .    ^
         negation is already there